common/djangoapps/cache_toolbox/middleware.py

```diff
@@ -106,6 +109,19 @@ def process_request(self, request):
                 )
                 # Raise an exception to fall through to the except clause below.
                 raise Exception
+            # Ensure that the user's session hash hasn't changed. We
+            # check that SessionAuthenticationMiddleware is enabled in
+            # order to match Django's behavior.
+            session_auth_class = 'django.contrib.auth.middleware.SessionAuthenticationMiddleware'
+            session_auth_enabled = session_auth_class in settings.MIDDLEWARE_CLASSES
+            auto_auth_enabled = settings.FEATURES.get('AUTOMATIC_AUTH_FOR_TESTING', False)
+            if not auto_auth_enabled and session_auth_enabled and hasattr(request.user, 'get_session_auth_hash'):
```

Comment:
Ah.  I see.  So you can either:
1. Add the session hash to the auto-auth User object in the [auto_auth](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/views.py#L1896) endpoint.
2. Special case auto-auth users in this method since the current implementation of auto-auth is only intended for low-security environments - which should not require this auto-logout feature.

And so you've implemented option #2.  That's fine.